flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/pattern/InvalidPatternException.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.cep.pattern;
+
+/**
+ * Thrown when the given {@link Pattern} was constructed illegaly e.g when
+ * more than quantifier provided.
+ */
+class InvalidPatternException extends RuntimeException {
```

Comment:
Instead of creating a new `Exception` I suggest we should use the existing `MalformedPatternException`. You can move it to this package, as it is used more broadly now.